{"nbformat":4,"nbformat_minor":0,"metadata":{"kernelspec":{"display_name":"Python 3","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.8.2"},"colab":{"name":"A_05_Errors_Exceptions_Handling_en.ipynb","provenance":[{"file_id":"1wF49MAUVV2NAR0xwJdKx1p_HR9PPYbSk","timestamp":1615383804000}],"collapsed_sections":[]}},"cells":[{"cell_type":"markdown","metadata":{"id":"QUor4QoDRpBw"},"source":["**(1)** Let us define a division operation inside a function (using `def`) but to get an error, define the denominator as 0. So, type properly except statement."]},{"cell_type":"code","metadata":{"id":"dQVYlKMdRpB3","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1615383883520,"user_tz":-180,"elapsed":855,"user":{"displayName":"Melih Can Yardı","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiDRpIoK6Mk9syMJ6ns-2M9sxIr_XQLtyHZbSnEUg=s64","userId":"14731427163213635722"}},"outputId":"54023370-6d69-40d4-f6c6-da570f591dce"},"source":["def division(divident, denominator):\r\n","  try:\r\n","    return divident / denominator\r\n","  except ZeroDivisionError:\r\n","    print(\"Denominator cannot be 0.\")\r\n","\r\n","print(division(25, 5))\r\n","division(25, 0)"],"execution_count":1,"outputs":[{"output_type":"stream","text":["5.0\n","Denominator cannot be 0.\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"VzLNjDpcRpB5"},"source":["**(2)** It is possible to get multiple errors after the execution of one try block. So, please define an error-exception issue with `NameError`."]},{"cell_type":"code","metadata":{"id":"aThaKak3RpB5","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1615383883884,"user_tz":-180,"elapsed":1194,"user":{"displayName":"Melih Can Yardı","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiDRpIoK6Mk9syMJ6ns-2M9sxIr_XQLtyHZbSnEUg=s64","userId":"14731427163213635722"}},"outputId":"11d4efdb-428a-4366-da1e-0e51cccefc1a"},"source":["try:\r\n","  print(\"Hello\" + name)\r\n","except NameError:\r\n","  print('The variable \"name\" is not defined.')"],"execution_count":2,"outputs":[{"output_type":"stream","text":["The variable \"name\" is not defined.\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"nUtqGS6pRpB6"},"source":["**(3)** Please define a function and with this function, generate a `ValueError` exception simply by entering a string instead of numerical value."]},{"cell_type":"code","metadata":{"id":"kJY-8updRpB6","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1615383883886,"user_tz":-180,"elapsed":1184,"user":{"displayName":"Melih Can Yardı","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiDRpIoK6Mk9syMJ6ns-2M9sxIr_XQLtyHZbSnEUg=s64","userId":"14731427163213635722"}},"outputId":"9a72e999-c5f2-4128-ddd1-1a19e8dc4aab"},"source":["def add_five(num):\r\n","  try:\r\n","    return int(num) + 5\r\n","  except:\r\n","    print(\"Please enter a number only!\")\r\n","  \r\n","print(add_five(5))\r\n","add_five('five')"],"execution_count":3,"outputs":[{"output_type":"stream","text":["10\n","Please enter a number only!\n"],"name":"stdout"}]}]}